src/peep/peep.c

```diff
@@ -3421,7 +3444,616 @@ static void peep_update_fixing(int steps, rct_peep* peep){
+				break;
+
+			case 14:
+				do_6c1653 = peep_update_fixing_sub_state_14(firstRun, peep, ride);
+				break;
+
+			default:
+				log_error("Invalid substate");
+				do_6c1653 = false;
+		}
+
+		firstRun = false;
+
+		if (!do_6c1653) {
+			break;
+		}
+
+		int edx = peep->sub_state;
+		uint32 ebp = RCT2_ADDRESS(0x992A18, uint32)[8];
+
+		if (peep->state != PEEP_STATE_INSPECTING) {
+			ebp = RCT2_ADDRESS(0x992A18, uint32)[ride->breakdown_reason_pending];
+		}
+
+		// loc_6C167B
+		do {
+			edx++;
+		} while ((ebp & (1 << edx)) == 0);
+
+		peep->sub_state = edx & 0xFF;
+	}
+}
+
+/**
+ * rct2: 0x006C0EEC
+ */
+static bool peep_update_fixing_sub_state_0(rct_ride *ride) {
+	ride->mechanic_status = RIDE_MECHANIC_STATUS_FIXING;
+	ride->window_invalidate_flags |= RIDE_INVALIDATE_RIDE_MAINTENANCE;
+
+	return true;
+}
+
+/**
+ * rct2: 0x006C0F09
+ */
+static bool peep_update_fixing_sub_state_1(bool firstRun, rct_peep *peep, rct_ride *ride) {
+	sint16 x, y, tmp_xy_distance;
+
+	if (!firstRun) {
+		uint16 vehicleIndex = ride->vehicles[ride->broken_vehicle];
+
+		if (vehicleIndex == 0xFFFF) {
+			return true;
+		}
+
+		rct_vehicle *vehicle = GET_VEHICLE(vehicleIndex);
+		for (uint8 i = 0; i < ride->broken_car; i++) {
+			vehicle = GET_VEHICLE(vehicle->next_vehicle_on_train);
+		}
+
+		while (true) {
+			if (vehicle->is_child == 0) {
+				break;
+			}
+
+			uint8 trackType = vehicle->track_type >> 2;
+			if (trackType == TRACK_ELEM_END_STATION) {
+				break;
+			}
+
+			if (trackType == TRACK_ELEM_BEGIN_STATION) {
+				break;
+			}
+
+			if (trackType == TRACK_ELEM_MIDDLE_STATION) {
+				break;
+			}
+
+			vehicle = GET_VEHICLE(vehicle->prev_vehicle_on_ride);
+		}
+
+		rct_xy16 offset = word_981D6C[peep->var_78];
+		peep->destination_x = (offset.x * -12) + vehicle->x;
+		peep->destination_y = (offset.y * -12) + vehicle->y;
+		peep->destination_tolerence = 2;
+	}
+
+	invalidate_sprite_2((rct_sprite *) peep);
+	if (peep_update_action(&x, &y, &tmp_xy_distance, peep)) {
+		sprite_move(x, y, peep->z, (rct_sprite *) peep);
+		invalidate_sprite_2((rct_sprite *) peep);
+		return false;
+	}
+
+	return true;
+}
+
+/**
+ * rct2: 0x006C0FD3
+ */
+static bool peep_update_fixing_sub_state_2345(bool firstRun, rct_peep *peep, rct_ride *ride) {
+	sint16 tmp_x, tmp_y, tmp_distance;
+
+	if (!firstRun) {
+		peep->sprite_direction = peep->var_78 << 3;
+
+		peep->action = PEEP_ACTION_STAFF_FIX;
+		if (scenario_rand() & 0x100000) {
+			peep->action = PEEP_ACTION_STAFF_FIX_2;
+		}
+
+		peep->action_sprite_image_offset = 0;
+		peep->action_frame = 0;
+		sub_693B58(peep);
+		invalidate_sprite_2((rct_sprite *) peep);
+	}
+
+	if (peep->action == 0xFF) {
+		return true;
+	}
+
+	peep_update_action(&tmp_x, &tmp_y, &tmp_distance, peep);
+
+	uint8 actionFrame = 0x25;
+	if (peep->action != 0x0F) {
+		actionFrame = 0x50;
+	}
+
+	if (peep->action_frame != actionFrame) {
+		return false;
+	}
+
+	uint16 vehicleIndex = ride->vehicles[ride->broken_vehicle];
```

Comment:
I said this before but think it might have been lost in the comments. Should probably just make a function to get the broken car since you use this a couple times.